spec.emu

```diff
@@ -52,13 +54,17 @@ contributors:
         </h1>
         <dl class="header">
           <dt>description</dt>
-          <dd>It returns a string representing a |Pattern| for matching _c_. If _c_ is white space or an ASCII punctuator, the returned value is an escape sequence (corresponding with |HexEscapeSequence| if possible, or otherwise with |RegExpUnicodeEscapeSequence|). Otherwise, the returned value is a string representation of _c_ itself.</dd>
+          <dd>It returns a string representing a |Pattern| for matching _c_. If _c_ is white space or an ASCII punctuator, the returned value is an escape sequence. Otherwise, the returned value is a string representation of _c_ itself.</dd>
         </dl>
 
         <emu-alg>
-          1. Let _punctuators_ be the string-concatenation of *"(){}[]|,.?\*+-^$=<>/#&!%:;@~'`"*, the code unit 0x0022 (QUOTATION MARK), and the code unit 0x005C (REVERSE SOLIDUS).
-          1. Let _toEscape_ be StringToCodePoints(_punctuators_).
-          1. If _toEscape_ contains _c_ or _c_ is matched by |WhiteSpace|, then
+          1. If _c_ is matched by |SyntaxCharacter| or _c_ is U+002F (SOLIDUS), then
+            1. Return the string-concatenation of 0x005C (REVERSE SOLIDUS) and UTF16EncodeCodePoint(_c_).
```

Comment:
This strikes me as a point-in-time snapshot, and I'm not a fan because it will be weird if e.g. `\@` becomes a valid escape in the future. My preference remains the universally applicable `\x…`/`\u…` approach. But that said, there is no technical issue here; merely an aesthetic one.